vtex-apps/place-components
# CHK-544: Add useAddressForm hook

#### What problem is this solving?

This will enable external components to fine tune the behavior of the `AddressForm` component. Before this we wouldn't be able to show the form errors programatically (because they are only shown when the user has blurred the input) such as when the user clicks a submit button with an invalid form.

#### How should this be manually tested?

[Workspace](https://addr--checkoutio.myvtex.com/cart/add?sku=289).

Steps to follow:

1. Proceed to checkout
2. Insert a first purchase email address
3. Fill your profile info
4. Use the `22250-040` postal code
5. Select any SLA and proceed to address
6. With the address form open, click the "go to payment" button
7. The UI should show an error message under the number input

#### Checklist/Reminders


#### Screenshots or example usage

https://user-images.githubusercontent.com/10223856/107513563-2a567c00-6b87-11eb-9b7e-2d057575718c.mov

#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements
